src/renderers/shaders/ShaderChunk/shadowmap_pars_fragment.glsl

```diff
@@ -1,12 +1,17 @@
 #ifdef USE_SHADOWMAP
 
+	uniform int isShadowCube[ MAX_SHADOWS ];
+	uniform samplerCube shadowCube[ MAX_SHADOWS ];
 	uniform sampler2D shadowMap[ MAX_SHADOWS ];
 	uniform vec2 shadowMapSize[ MAX_SHADOWS ];
 
 	uniform float shadowDarkness[ MAX_SHADOWS ];
 	uniform float shadowBias[ MAX_SHADOWS ];
 
+	uniform vec3 shadowLightPosition[ MAX_SHADOWS ];
+
 	varying vec4 vShadowCoord[ MAX_SHADOWS ];
```

Comment:
Ah, so are you thinking of doing away with 'shadowMatrix'? I thought it would be OK to use it in the alternate manner that I described above since it wouldn't add any extra computations over the existing shadow mapping code (in the vertex shader). Basically the vertex shader would look something like:

```
//  shadowDarkness[ i ] < 0 means we have a point light
if( shadowDarkness[ i ] < 0 ){

     vShadowCoord[ i ] = shadowMatrix[ i ] * vec4( pointLightPosition[ i ], 1.0 );

} else {

     vShadowCoord[ i ] = shadowMatrix[ i ] * worldPosition;

}
```

Additionally, I would be re-purposing 'vShadowCoord' to hold the world-space position of the light. I admit it's not the most readable code, but I'd make sure to comment appropriately.